### ISSUE TITLE
Error: bcftools sort  returned non-zero exit status 255

### ISSUE THREAD

**Issue by carmensandoval** [wheaton5/souporcell#199]
I'm getting a strange error related to bcftools sort, any ideas what this could be due to?

Thank you.

```
#! /bin/bash

set -euxo pipefail
ml load singularity

demuxafy_dir="/path/to/demuxafy"
data_dir="/path/to/data"

singularity exec \
-B "$demuxafy_dir" \
-B "$data_dir" \
$demuxafy_dir/Demuxafy.sif \
souporcell_pipeline.py \
-t 8 \
-f $demuxafy_dir/refdata-gex-GRCh38-2020-A/fasta/genome.fa \
-i $data_dir/souporcell/test/possorted_genome_bam_subsampled.bam \
-b $data_dir/souporcell/test/subsampled_barcodes.tsv \
-o $data_dir/souporcell/test/demuxafy_out/SAM24434291 \
-k 3 \
```

```
running freebayes
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r KI270442.1:0-392061
freebayes -f/path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr11:1721089-135086622
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr14:6463670-107043718
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr18:11301817-80373285
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r KI270727.1:0-448248
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r KI270728.1:0-1872759
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r KI270729.1:0-280839
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr10:0-133797422
.
.
.
.
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr8:0-117819804
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr13:0-114364328
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa  -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr17:0-83257441
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa  -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chrY:0-57227415
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa  -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr6:0-60502916
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa  -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr4:0-44786890
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa  -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr18:0-11301817
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa  -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr20:0-17586227
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa  -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr11:0-1721089
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa  -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chr14:0-6463670

merging vcfs
Checking the headers and starting positions of 35 files
Traceback (most recent call last):
  File "/opt/souporcell/souporcell_pipeline.py", line 586, in <module>
    final_vcf = freebayes(args, bam, fasta)
  File "/opt/souporcell/souporcell_pipeline.py", line 474, in freebayes
    subprocess.check_call(['bcftools', 'sort', args.out_dir + "/souporcell_merged_vcf.vcf"], stdout = vcfout, stderr = vcferr)
  File "/opt/conda/envs/py36/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['bcftools', 'sort', 'path/to/souporcell/test/demuxafy_out/SAM24434291/souporcell_merged_vcf.vcf']' returned non-zero exit status 255.
```

**Comment by wheaton5** [comment]
Any output in the .err files?

**Comment by carmensandoval** [comment]
`bcftools.err` says: `mkdtemp(/local/13263069/bcftools.wJjUTT) failed: No such file or directory`

**Comment by wheaton5** [comment]
Probably not the demuxify container as it probably calls right into my container.

**Comment by wheaton5** [comment]
can I get an ls on the output directory?

**Comment by wheaton5** [comment]
sorry, that is the output of that process, so souporcell_merged_vcf.vcf does not exist

**Comment by wheaton5** [comment]
so its possible that this command failed
subprocess.check_call(["bcftools", "concat", '-a'] + all_vcfs, stdout = vcfout)
but I don't pipe the stderr to a file

**Comment by carmensandoval** [comment]
out dir:

```
total 755887
drwxr-xr-x 2 sandovc9 idmgmt     32768 Aug 14 18:16 .
drwxr-xr-x 3 sandovc9 idmgmt      8192 Aug 14 15:56 ..
-rw-r--r-- 1 sandovc9 idmgmt        75 Aug 14 18:16 bcftools.err
-rw-r--r-- 1 sandovc9 idmgmt     23554 Aug 14 17:56 fastqs.done
-rw-r--r-- 1 sandovc9 idmgmt      1406 Aug 14 18:12 minimap.err
-rw-r--r-- 1 sandovc9 idmgmt       968 Aug 14 18:12 remapping.done
-rw-r--r-- 1 sandovc9 idmgmt         0 Aug 14 18:13 retag.err
-rw-r--r-- 1 sandovc9 idmgmt         0 Aug 14 18:14 retagging.done
-rw-r--r-- 1 sandovc9 idmgmt       342 Aug 14 18:16 souporcell_0_0.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      3486 Aug 14 18:16 souporcell_0_0.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt        79 Aug 14 18:16 souporcell_0_0.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       342 Aug 14 18:16 souporcell_0_1.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      3486 Aug 14 18:16 souporcell_0_1.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt        79 Aug 14 18:16 souporcell_0_1.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       343 Aug 14 18:16 souporcell_0_2.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      3486 Aug 14 18:16 souporcell_0_2.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt        79 Aug 14 18:16 souporcell_0_2.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       342 Aug 14 18:16 souporcell_0_3.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      3486 Aug 14 18:16 souporcell_0_3.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt        79 Aug 14 18:16 souporcell_0_3.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_0_4.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     42347 Aug 14 18:16 souporcell_0_4.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      3214 Aug 14 18:16 souporcell_0_4.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       340 Aug 14 18:16 souporcell_0_5.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     24381 Aug 14 18:16 souporcell_0_5.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1726 Aug 14 18:16 souporcell_0_5.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       338 Aug 14 18:16 souporcell_0_6.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      5565 Aug 14 18:16 souporcell_0_6.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       200 Aug 14 18:16 souporcell_0_6.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       346 Aug 14 18:16 souporcell_1_0.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     28669 Aug 14 18:16 souporcell_1_0.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      2060 Aug 14 18:16 souporcell_1_0.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       340 Aug 14 18:16 souporcell_1_1.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     31692 Aug 14 18:16 souporcell_1_1.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      2202 Aug 14 18:16 souporcell_1_1.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       340 Aug 14 18:16 souporcell_1_2.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     22454 Aug 14 18:16 souporcell_1_2.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1400 Aug 14 18:16 souporcell_1_2.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       338 Aug 14 18:16 souporcell_1_3.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      3486 Aug 14 18:16 souporcell_1_3.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt        79 Aug 14 18:16 souporcell_1_3.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       346 Aug 14 18:16 souporcell_2_0.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     21392 Aug 14 18:16 souporcell_2_0.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1525 Aug 14 18:16 souporcell_2_0.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       340 Aug 14 18:16 souporcell_2_1.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     22812 Aug 14 18:16 souporcell_2_1.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1341 Aug 14 18:16 souporcell_2_1.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_2_2.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     24687 Aug 14 18:16 souporcell_2_2.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1503 Aug 14 18:16 souporcell_2_2.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_2_3.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     25468 Aug 14 18:16 souporcell_2_3.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1732 Aug 14 18:16 souporcell_2_3.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_2_4.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      6597 Aug 14 18:16 souporcell_2_4.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       276 Aug 14 18:16 souporcell_2_4.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       346 Aug 14 18:16 souporcell_3_0.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      8707 Aug 14 18:16 souporcell_3_0.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       555 Aug 14 18:16 souporcell_3_0.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_3_1.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     24832 Aug 14 18:16 souporcell_3_1.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1656 Aug 14 18:16 souporcell_3_1.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_3_2.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     48017 Aug 14 18:16 souporcell_3_2.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      3598 Aug 14 18:16 souporcell_3_2.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_3_3.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      8083 Aug 14 18:16 souporcell_3_3.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       431 Aug 14 18:16 souporcell_3_3.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       346 Aug 14 18:16 souporcell_4_0.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     10683 Aug 14 18:16 souporcell_4_0.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       750 Aug 14 18:16 souporcell_4_0.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_4_1.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      8090 Aug 14 18:16 souporcell_4_1.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       459 Aug 14 18:16 souporcell_4_1.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_4_2.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     13912 Aug 14 18:16 souporcell_4_2.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       813 Aug 14 18:16 souporcell_4_2.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_4_3.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     42440 Aug 14 18:16 souporcell_4_3.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      2849 Aug 14 18:16 souporcell_4_3.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       338 Aug 14 18:16 souporcell_4_4.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     28855 Aug 14 18:16 souporcell_4_4.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1051 Aug 14 18:16 souporcell_4_4.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       346 Aug 14 18:16 souporcell_5_0.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     20332 Aug 14 18:16 souporcell_5_0.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1313 Aug 14 18:16 souporcell_5_0.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_5_1.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     29982 Aug 14 18:16 souporcell_5_1.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      2189 Aug 14 18:16 souporcell_5_1.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       338 Aug 14 18:16 souporcell_5_2.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     11238 Aug 14 18:16 souporcell_5_2.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       711 Aug 14 18:16 souporcell_5_2.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       346 Aug 14 18:16 souporcell_6_0.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     21201 Aug 14 18:16 souporcell_6_0.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1396 Aug 14 18:16 souporcell_6_0.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_6_1.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     30357 Aug 14 18:16 souporcell_6_1.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      2185 Aug 14 18:16 souporcell_6_1.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_6_2.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     27239 Aug 14 18:16 souporcell_6_2.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1652 Aug 14 18:16 souporcell_6_2.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       347 Aug 14 18:16 souporcell_7_0.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      7357 Aug 14 18:16 souporcell_7_0.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       418 Aug 14 18:16 souporcell_7_0.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_7_1.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     21546 Aug 14 18:16 souporcell_7_1.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1531 Aug 14 18:16 souporcell_7_1.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       339 Aug 14 18:16 souporcell_7_2.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt     18892 Aug 14 18:16 souporcell_7_2.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt      1156 Aug 14 18:16 souporcell_7_2.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt       338 Aug 14 18:16 souporcell_7_3.vcf.err
-rw-r--r-- 1 sandovc9 idmgmt      4089 Aug 14 18:16 souporcell_7_3.vcf.gz
-rw-r--r-- 1 sandovc9 idmgmt       144 Aug 14 18:16 souporcell_7_3.vcf.gz.csi
-rw-r--r-- 1 sandovc9 idmgmt         0 Aug 14 18:16 souporcell_merged_sorted_vcf.vcf
-rw-r--r-- 1 sandovc9 idmgmt   2764868 Aug 14 18:16 souporcell_merged_vcf.vcf
-rw-r--r-- 1 sandovc9 idmgmt 766298438 Aug 14 18:14 souporcell_minimap_tagged_sorted.bam
-rw-r--r-- 1 sandovc9 idmgmt   3710496 Aug 14 18:14 souporcell_minimap_tagged_sorted.bam.bai
```

**Comment by wheaton5** [comment]
thanks. thinking...

**Comment by wheaton5** [comment]
i assume all of those .vcf.err files are empty? this might be a hard one. it might even be the singularity thing if we use different shells or something... not sure.

**Comment by carmensandoval** [comment]
`souporcell_merged_vcf.vcf`[ looks fine](https://gist.github.com/carmensandoval/bd3b5f0b2b3e1c996a0975a014d642d6), but `souporcell_merged_sorted_vcf.vcf` is empty.

I worry that the GRCh38 reference I'm using might be weird? With all those extra contigs? It's just the one we use by default at my org...

And the .vcf.err files just have the command, eg:

```
freebayes -f /path/to/demuxafy/refdata-gex-GRCh38-2020-A/fasta/genome.fa -iXu -C 2 -q 20 -n 3 -E 1 -m 30 --min-coverage 20 --pooled-continuous --skip-coverage 100000 -r chrX:0-156040895 /path/to/souporcell/test/demuxafy_out/SAM24434291/souporcell_minimap_tagged_sorted.bam
```

**Comment by wheaton5** [comment]
i use the same stupid grch38 reference with tons of extra contigs for a lot of my work. I hate it, but it should be fine.

**Comment by wheaton5** [comment]
still thinking

**Comment by carmensandoval** [comment]
Alright - I'm going to try with a different dataset and see how that goes...
FYI it also failed with a `--common_variants` file. I used the GRCh38 vcf you provide in the docs, along with this same `refdata-gex-GRCh38-2020-A` reference, and at first I was worried it had failed because of the non-overlapping contigs, but I'm guessing that wasn't it either, since you presumably use that combo as well.

Thanks for your help! Hopefully I can figure something else out...

**Comment by wheaton5** [comment]
going back to your original response on .err files
```
mkdtemp(/local/13263069/bcftools.wJjUTT) failed: No such file or directory
```
are you possibly running out of storage? bcftools seems to want to make a temp directory and is unable to

**Comment by carmensandoval** [comment]
I should be getting an email if I'm near of at my quota, so I don't think so, but I can try writing the output to my scratch dir and see if that helps!

**Comment by wheaton5** [comment]
check your usage and if it is a user limited system, maybe clear up data. also maybe in case it is trying to create tmp storage globally and you need local tmp storage, I think you can set your tmp directory export TMPDIR=/home/whatever (you would need to either run this before starting the run or add it to your ~/.bashrc or ~/.bash_profile and source ~/.bashrc or source ~/.bash_profile in the terminal before running

**Comment by wheaton5** [comment]
ok ill let you try things out for now. I think that's all of the insight I can bring to bare for now. I know ppl have been able to run through the demuxafy container before.

**Comment by carmensandoval** [comment]
OK so if I understand corrctly, I should:

```
> export TMPDIR="/path/to/scratch/u/sandovc9"

# Confirming
> echo $TMPDIR 
/path/to/scratch/u/sandovc9
```
and then I'd still need to bind that $TMPDIR in the singularity command?

```
singularity exec \
-B "$demuxafy_dir" \
-B "$data_dir" \
-B "$cellranger_out" \
-B "$TMPDIR" \
$demuxafy_dir/Demuxafy.sif \
souporcell_pipeline.py \
-t 8 \
-f $demuxafy_dir/refdata-gex-GRCh38-2020-A/fasta/genome.fa \
-i $data_dir/souporcell/test/possorted_genome_bam_subsampled.bam \
-b $data_dir/souporcell/test/subsampled_barcodes.tsv \
-o $data_dir/souporcell/test/demuxafy_out/SAM24434291 \
-k 3 \
```

**Comment by wheaton5** [comment]
That seems right, but im not very experienced in this.

**Comment by carmensandoval** [comment]
Thanks fro your help! Indeed it was an issue with `$TMPDIR`, and setting it before starting the run, and binding it in the singularity command did the trick.

I am able to complete the run _without_ a common variants file now, but still experiencing issues with one. But as this is a separate issue it seems others have had as well, I'll move over to that one.

**Comment by wheaton5** [comment]
seems resolved, closing
